cs262software/Back-End
# amc49/plays/lines

**0.11 B**
Create an endpoint that, given a play, an act, and a scene, returns the lineIDs, LineNums, and Text that are in that scene as a response.

**This pull request addresses a few other changes:**

- Rename routes to use '/api/...' to interface with the Front-End
- Add "*Passwords.js" to .gitignore to keep passwords out of github
- Rename "scripts" routes and files to "plays" to keep consitency with the database naming scheme (hence all of the marked deletions)

**To test:**

1. Checkout branch, run `npm install`, and don't forget to make a`modelPasswords.js` file (You need to have granted all permisions to 'theatreSuiteUser'@'localhost' in your database. It's password goes in modelPasswords.js).
2. Run `npm start` to start the server
3. Check for invalid parameters. For example, go to *http://localhost:3001/api/plays/lines*, which should give an error response.
4. Add a line or two to your database (you'll also need at least one play for this to work). For example, run:

`INSERT INTO play (PlayID, Name, FilePath) VALUES (1, "test123", "C:/afakepath/test123");
INSERT INTO line (LineID, PlayID, ActNum, SceneNum, LineNum, Text, DirectorNote) VALUES (1, 1, 1, 1, 1, "Once upon a time...", "Note: This is a fake play.");`

5. Start the server again and go to *http://localhost:3001/api/plays/lines?playID=1&actNum=1&sceneNum=1*, which should return the play that you just added
6. Go ahead and try other tests that you can think of.
